### PR TITLE
Use Metrics project

### PR DESCRIPTION
**Pull request by dfuchss**
Replace ArDoCo's Metric calculation with our Metrics project

Changes:
* We store the complete metrics
* We use micro average per project for MME

### CODE REVIEW

File: tests/integration-tests/tests-base/src/main/java/edu/kit/kastel/mcse/ardoco/core/tests/TestUtil.java [hunk 1]
```diff
@@ -37,20 +34,18 @@ public static <T> EvaluationResults<T> compareTLR(ArDoCoResult arDoCoResult, Imm
...
+        int sentences = arDoCoResult.getText().getSentences().size();
+        int modelElements = 0;
+        for (var model : arDoCoResult.getModelIds()) {
+            modelElements += arDoCoResult.getModelState(model).getInstances().size();
+        }
 
-        // False Positives are the trace links that are only contained in the result set
-        Set<T> falsePositives = distinctTraceLinks.stream().filter(tl -> !distinctGoldStandard.contains(tl)).collect(Collectors.toSet());
-        ImmutableList<T> falsePositivesList = Lists.immutable.ofAll(falsePositives);
+        int confusionMatrixSum = sentences * modelElements;
```
Comment (dfuchss) [inline tests/integration-tests/tests-base/src/main/java/edu/kit/kastel/mcse/ardoco/core/tests/TestUtil.java:43]: Please Check

File: tests/integration-tests/tests-base/src/main/java/edu/kit/kastel/mcse/ardoco/core/tests/TestUtil.java [hunk 2]
```diff
@@ -67,57 +62,11 @@ public static <T> EvaluationResults<T> compareInconsistencies(ArDoCoResult arDoC
...
-     * @param arDoCoResult   the output of ArDoCo
-     * @param truePositives  nr of true positives
-     * @param falsePositives nr of false positives
-     * @param falseNegatives nr of false negatives
-     * @return the number of true negatives
-     */
-    public static int calculateTrueNegativesForInconsistencies(ArDoCoResult arDoCoResult, int truePositives, int falsePositives, int falseNegatives) {
-        int numberOfSentences = arDoCoResult.getText().getSentences().size();
-        return numberOfSentences - (truePositives + falsePositives + falseNegatives);
+        int confusionMatrixSum = arDoCoResult.getText().getSentences().size();
```
Comment (dfuchss) [inline tests/integration-tests/tests-base/src/main/java/edu/kit/kastel/mcse/ardoco/core/tests/TestUtil.java:65]: Please Check

File: tests/integration-tests/tests-inconsistency/src/test/java/edu/kit/kastel/mcse/ardoco/id/tests/integration/InconsistencyDetectionEvaluationIT.java [hunk 3]
```diff
@@ -529,18 +528,15 @@ private static Pair<MutableList<EvaluationResults<String>>, StringBuilder> inspe
...
         appendResults(falsePositives, detailedOutputBuilder, "False Positives", arDoCoResult, outputBuilder);
 
-        var falseNegatives = result.falseNegatives().toList();
+        var falseNegatives = result.falseNegatives();
         appendResults(falseNegatives, detailedOutputBuilder, "False Negatives", arDoCoResult, outputBuilder);
-
-        var results = EvaluationResults.createEvaluationResults(new ResultMatrix<>(truePositives.toImmutable(), 0, falsePositives.toImmutable(), falseNegatives
-                .toImmutable()));
-        allResults.add(results);
+        allResults.add(result);
```
Comment (dfuchss) [inline tests/integration-tests/tests-inconsistency/src/test/java/edu/kit/kastel/mcse/ardoco/id/tests/integration/InconsistencyDetectionEvaluationIT.java:539]: Please check. I now use the original result for serialization. Not the one with reduced information.

**Review by dfuchss — COMMENTED** [review]
Some comments to the PR

File: tests/integration-tests/tests-tlr/src/test/java/edu/kit/kastel/mcse/ardoco/tlr/tests/integration/tlrhelper/ModelElementSentenceLink.java [hunk 4]
```diff
@@ -8,15 +8,15 @@
 /**
  * Represents a simple trace link by the id of the model and number of the sentence involved.
  */
-public record TestLink(String modelId, int sentenceNr) implements Comparable<TestLink> {
+public record ModelElementSentenceLink(String modelElementId, int sentenceNr) implements Comparable<ModelElementSentenceLink> {
```
Comment (Gram21) [inline tests/integration-tests/tests-tlr/src/test/java/edu/kit/kastel/mcse/ardoco/tlr/tests/integration/tlrhelper/ModelElementSentenceLink.java:11]: Was this way before, but maybe we can directly clean up:
I suggest to rename the `sentenceNr` in `sentenceNumber`

**Review by Gram21 — APPROVED** [review]
Overall good changes with only one direct minor comment.
As we shift a lot of responsibility to the dependency ArDoCo/Metric, we need to thoroughly review it as well. Given that the results are overall the same (with only some changes where we deliberately changed from weighted to micro averages), I guess it is initially okay.

**Pull request merged**